tron/kubernetes.py

```diff
@@ -292,23 +319,78 @@ def create_task(self, action_run_id: str, serializer, task_id: Optional[str] = N
             )
             return None
 
-        # TODO: fill out required fields once they're ready
-        task_config = self.runner.TASK_CONFIG_INTERFACE()
-
+        task_config = cast(
+            KubernetesTaskConfig,
+            self.runner.TASK_CONFIG_INTERFACE(
+                name=action_run_id,
+                command=command,
+                image=docker_image,
+                cpus=cpus,
+                memory=mem,
+                disk=DEFAULT_DISK_LIMIT if disk is None else disk,
+                environment=env,
+                volumes=[
+                    volume._asdict()
+                    for volume in combine_volumes(defaults=self.default_volumes or [], overrides=volumes)
+                ],
+            ),
+        )
+
+        # this should only ever be non-null when we're recovering from a Tron restart
+        # and are recreating the previous state - when actually creating a new task
+        # we'll always let task_processing come up with a Pod name for us
         if task_id is not None:
             try:
                 task_config = task_config.set_pod_name(task_id)
             except ValueError:
                 log.error(f"Invalid {task_id} for {action_run_id}")
                 return None
 
-        return KubernetesTask(action_run_id=action_run_id, task_config=task_config, serializer=serializer)
+        return KubernetesTask(action_run_id=action_run_id, task_config=task_config, serializer=serializer,)
+
+    def _check_connection(self) -> None:
+        """
+        Helper to ensure that the task_processing plugin is in a running state and event handling
+        is correctly setup in case we've disabled k8s at some point during operation.
+        """
+        if self.runner is None or self.runner.stopping:
+            log.info("k8s plugin never created or stopped, restarting.")
+            self.connect()
+        # re-add callbacks just in case they're missing
+        elif self.deferred is None or self.deferred.called:
```

Comment:
should this be an elif or just another if? Otherwise if we did have to reconnect above, we'd never re-check callbacks here

Reply:
if we did have to reconnect, that path will eventually also call `self.handle_next_event` and do this same check before calling `self.handle_next_event`